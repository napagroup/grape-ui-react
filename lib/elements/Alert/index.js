"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _component = require("./component");

_forEachInstanceProperty(_context = _Object$keys(_component)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _component[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});