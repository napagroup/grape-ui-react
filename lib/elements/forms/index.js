"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2, _context3, _context4, _context5, _context6;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _CheckboxField = require("./CheckboxField");

_forEachInstanceProperty(_context = _Object$keys(_CheckboxField)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _CheckboxField[key];
    }
  });
});

var _DateField = require("./DateField");

_forEachInstanceProperty(_context2 = _Object$keys(_DateField)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _DateField[key];
    }
  });
});

var _Form = require("./Form");

_forEachInstanceProperty(_context3 = _Object$keys(_Form)).call(_context3, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Form[key];
    }
  });
});

var _RadioField = require("./RadioField");

_forEachInstanceProperty(_context4 = _Object$keys(_RadioField)).call(_context4, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _RadioField[key];
    }
  });
});

var _SelectField = require("./SelectField");

_forEachInstanceProperty(_context5 = _Object$keys(_SelectField)).call(_context5, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _SelectField[key];
    }
  });
});

var _TextField = require("./TextField");

_forEachInstanceProperty(_context6 = _Object$keys(_TextField)).call(_context6, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TextField[key];
    }
  });
});