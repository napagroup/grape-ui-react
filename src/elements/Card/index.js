export * from './component';
export * from './styled';
