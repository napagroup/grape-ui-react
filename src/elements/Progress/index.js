export * from './getProgress';
export * from './styled';
