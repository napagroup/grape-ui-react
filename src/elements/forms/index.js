export * from './CheckboxField';
export * from './DateField';
export * from './Form';
export * from './SelectField';
export * from './TextField';
