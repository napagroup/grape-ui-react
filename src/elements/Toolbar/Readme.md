Toolbar is grape-ui's take on the [AppBar found in Material Design](https://material.io/components/app-bars-top). It can be used to help define navigation for your site. Toolbar can take direct children, however there are shortcuts for defining content to appear in different areas of the toolbar. Use these props to achieve it:

* `leftArea`
* `centerArea`
* `rightArea`

#### Basic Examples

```jsx in Markdown

<Toolbar>
  Plain Toolbar
</Toolbar>

<Toolbar
  background="cornflowerblue"
>
  Using background color.
</Toolbar>

<Toolbar
  leftArea="I'm on the left"
  centerArea="I'm in the middle"
  rightArea="I'm on the right"
>
  I'm the rest of the children!
</Toolbar>

```

#### Material Example

```jsx in Markdown
import styled, { ThemeProvider } from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons'
import { Button } from 'src/elements/Button';
import { TextField } from 'src/elements/forms';
import { Flex } from 'src/elements/grid';
import { Header } from 'src/elements/typography';

const ToolbarButton = styled(Button)``;

ToolbarButton.defaultProps = {
  bgActiveColor: 'purple.light',
  bgHoverColor: 'purple',
  color: 'white'
};

const leftArea = (
  <Flex alignItems="center" ml={-3}>
    <ToolbarButton>
      <FontAwesomeIcon icon={faBars} />
    </ToolbarButton>
    <Header.h5
      color="white"
      mb="0"
    >
      App
    </Header.h5>
  </Flex>
);

const rightArea = (
  <ToolbarButton>
    Login
  </ToolbarButton>
);

<ThemeProvider
  theme={{
    colors: {
      brandPrimary: 'rgb(203, 39, 168)',
    },
  }}
>
  <Toolbar
    bg="brandPrimary"
    leftArea={leftArea}
    rightArea={rightArea}
  />
</ThemeProvider>

```

#### This Styleguide's Toolbar

```jsx in Markdown
import { ReactSVG } from 'react-svg';
import { faGithub, faTwitter } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Button } from 'src/elements/Button';
import { Box, Flex } from 'src/elements/grid';
import { Link, Text } from 'src/elements/typography';
import logo from 'src/assets/images/grape-ui-header-logo.svg'

const leftArea = (
  <Flex alignItems="center" flexDirection={['column', 'row']}>
    <Box maxWidth={175}>
      <Link>
        <ReactSVG src={logo} />
      </Link>
    </Box>
    <Link>
      <Button color="black" fontWeight="bold">DOCS</Button>
    </Link>
  </Flex>
);

const rightArea = (version) => (
  <Flex alignItems="center">
    <Box mx={1} px={[1, 2, 3]}>
      <Text>v{version}</Text>
    </Box>
    <Button pl={[1, 2]} pr={[1, 2]}>
      <FontAwesomeIcon icon={faGithub} />
    </Button>
    <Button pl={[1, 2]} pr={[1, 2]}>
      <FontAwesomeIcon icon={faTwitter} />
    </Button>
    <Box display={['none', 'block']}>
      <Link>
        <Button color="brandPrimary">Getting Started</Button>
      </Link>
    </Box>
  </Flex>
);

<Toolbar
  borderBottom="1px solid rgb(203, 39, 168)"
  boxShadow="rgb(132, 11, 85) 0px 1px 0px"
  leftArea={leftArea}
  minHeight={null}
  px={[2, 3, 4, 5]}
  py={2}
  rightArea={rightArea('#.##')}
  toolbarInnerProps={{
    alignItems: 'center',
    flexDirection: ['column', 'row'],
    maxWidth: 'none',
    width: 1,
  }}
/>

```

#### Hide Toolbar

```jsx in Markdown
import { useState } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons'
import { Button } from 'src/elements/Button';
import { TextField } from 'src/elements/forms';
import { Flex } from 'src/elements/grid';
import { Header } from 'src/elements/typography';

const ToolbarButton = styled(Button)``;

ToolbarButton.defaultProps = {
  bgActiveColor: 'purple.light',
  bgHoverColor: 'purple',
  color: 'white'
};

const leftArea = (
  <Flex alignItems="center" ml={-3}>
    <ToolbarButton>
      <FontAwesomeIcon icon={faBars} />
    </ToolbarButton>
    <Header.h5
      color="white"
      mb="0"
    >
      App
    </Header.h5>
  </Flex>
);

const rightArea = (
  <ToolbarButton>
    Login
  </ToolbarButton>
);

const [hide, setHidden] = useState(false);

<ThemeProvider
  theme={{
    colors: {
      brandPrimary: 'rgb(203, 39, 168)',
    },
  }}
>
  <Button
    onClick={() => setHidden(!hide)}
  >
    Toggle Visibility
  </Button>
  <Toolbar
    bg="brandPrimary"
    isHidden={hide}
    leftArea={leftArea}
    rightArea={rightArea}
  />
</ThemeProvider>

```
