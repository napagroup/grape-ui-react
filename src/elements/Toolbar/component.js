import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Box, Flex } from 'src/elements/grid';
import { removeSomeProps } from 'src/utils/componentHelpers';
import { toolbarDefaultProps, toolbarPropTypes } from './utils';

const ToolbarInner = styled(Flex)``;

export const ToolbarComponent = ({
  centerArea,
  centerAreaProps,
  children,
  containerComponent,
  leftArea,
  leftAreaProps,
  rightArea,
  rightAreaProps,
  toolbarInnerProps,
  ...props
}) => {
  const {
    progress,
    progressPlacement,
    progressProps,
    showProgress,
  } = props;
  const boxProgressProps = {
    progress,
    progressPlacement,
    progressProps,
    showProgress,
  };
  return (
    <Box {...boxProgressProps}>
      <Box
        as={containerComponent}
        {...removeSomeProps(props, Object.keys(toolbarPropTypes))}
      >
        <ToolbarInner {...toolbarInnerProps}>
          {children}
          <Box {...leftAreaProps}>
            {leftArea}
          </Box>
          <Box {...centerAreaProps}>
            {centerArea}
          </Box>
          <Box {...rightAreaProps}>
            {rightArea}
          </Box>
        </ToolbarInner>
      </Box>
    </Box>
  );
};

ToolbarComponent.propTypes = {
  children: PropTypes.any,
  ...toolbarPropTypes,
};

ToolbarComponent.defaultProps = {
  children: null,
  ...toolbarDefaultProps,
};
