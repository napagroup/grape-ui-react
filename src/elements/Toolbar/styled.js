import styled from 'styled-components';
import {
  background,
  border,
  color,
  flexbox,
  layout,
  position,
  shadow,
  space,
  system,
} from 'styled-system';
import { withHideable } from 'src/elements/utils/Hideable';
import { ToolbarComponent } from './component';
import { toolbarDefaultProps, toolbarPropTypes } from './utils';

const Toolbar = styled(withHideable(ToolbarComponent))`
  ${system({ boxSizing: true })}
  ${background}
  ${border}
  ${color}
  ${flexbox}
  ${layout}
  ${position}
  ${shadow}
  ${space}
`;

Toolbar.propTypes = {
  ...toolbarPropTypes,
};

Toolbar.defaultProps = {
  ...toolbarDefaultProps,
};

/** @component */
export { Toolbar };
