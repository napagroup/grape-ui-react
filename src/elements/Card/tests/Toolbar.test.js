import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Toolbar } from 'src/elements/Toolbar';

const emptyTheme = {};

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};

describe('Box Component base', () => {
  it('should return an empty Toolbar with base styling', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Toolbar />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
