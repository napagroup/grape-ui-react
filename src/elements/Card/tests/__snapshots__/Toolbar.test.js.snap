// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component base should return an empty Toolbar with base styling 1`] = `
.c2 {
  box-sizing: border-box;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 960px;
  width: 100%;
}

.c0 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 48px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.12),0 1px 5px 0 rgba(0,0,0,0.2);
  padding-left: 16px;
  padding-right: 16px;
}

@media screen and (min-width:40em) {
  .c0 {
    min-height: 56px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    min-height: 64px;
  }
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<header
  className="c0"
>
  <div
    className="sc-AxirZ c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
    <div
      className="c2"
    />
  </div>
</header>
`;
